drivers/display/display_ili9340.c

```diff
@@ -1,154 +1,152 @@
 /*
  * Copyright (c) 2017 Jan Van Winkel <jan.van_winkel@dxplore.eu>
  * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Pavlo Hamov <pasha.gamov@gmail.com>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include "display_ili9340.h"
 #include <drivers/display.h>
 
-#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
-#include <logging/log.h>
-LOG_MODULE_REGISTER(display_ili9340);
-
 #include <drivers/gpio.h>
 #include <sys/byteorder.h>
 #include <drivers/spi.h>
 #include <string.h>
 
-struct ili9340_data {
-#ifdef DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_CONTROLLER
-	struct device *reset_gpio;
-#endif
-	struct device *command_data_gpio;
-	struct device *spi_dev;
-	struct spi_config spi_config;
-#ifdef DT_INST_0_ILITEK_ILI9340_CS_GPIOS_CONTROLLER
-	struct spi_cs_control cs_ctrl;
-#endif
-};
-
-#define ILI9340_CMD_DATA_PIN_COMMAND 1
-#define ILI9340_CMD_DATA_PIN_DATA 0
+#include <logging/log.h>
+LOG_MODULE_REGISTER(display_ili9340, CONFIG_DISPLAY_LOG_LEVEL);
 
-/* The number of bytes taken by a RGB pixel */
-#ifdef CONFIG_ILI9340_RGB565
-#define ILI9340_RGB_SIZE 2U
+#if (DT_INST_0_ILITEK_ILI9XXX_TYPE_ENUM == 0)
```

Comment:
Please remove this code `DT_INST_0_ILITEK_ILI9XXX_TYPE_ENUM` can never be 0 as it is required device parameter.

Reply:
As far as i understood how enums in dts works they are numerated in the same way as in C. 
0 - stand for the firts item in the dts enum (which corresponds to ili9163 for this case).
But this is too implicit. string comparison wont work for #if statement. I will try to investiagate what could be done with this magic number

Reply:
OK I did not know that the DTS enums returned numbers instead of strings, forget the remark